packages/versioning/src/versioning.ts

```diff
@@ -78,18 +89,48 @@ export function $madeOptional(context: DecoratorContext, t: Type, v: EnumMember)
   program.stateMap(madeOptionalKey).set(t, version);
 }
 
+function toVersion(p: Program, t: Type, v: ObjectType): Version | undefined {
+  const [namespace] = getVersions(p, t);
+  if (namespace === undefined) {
+    return undefined;
+  }
+  return getVersionForNamespace(
+    p,
+    v,
+    (namespace.projectionBase as Namespace) ?? namespace
+  ) as Version;
+}
+
+function getRenamedFrom(p: Program, t: Type): Array<RenamedFrom> | undefined {
+  return p.stateMap(renamedFromKey).get(t) as Array<RenamedFrom>;
+}
+
 /**
- * @returns version when the given type was added if applicable.
+ * @returns the list of versions for which this decorator has been applied
  */
-export function getRenamedFromVersion(p: Program, t: Type): Version | undefined {
-  return p.stateMap(renamedFromKey).get(t)?.v;
+export function getRenamedFromVersion(p: Program, t: Type): Array<Version> | undefined {
+  return getRenamedFrom(p, t)?.map((x) => x.v);
 }
 
 /**
  * @returns get old renamed name if applicable.
  */
-export function getRenamedFromOldName(p: Program, t: Type): string {
-  return p.stateMap(renamedFromKey).get(t)?.oldName ?? "";
+export function getRenamedFromOldName(p: Program, t: Type, v: ObjectType): string {
```

Comment:
I woudl change the name of this too
```suggestion
export function getNameAtVersion(p: Program, t: Type, v: ObjectType): string {
```

Reply:
Same here--renaming this would be a breaking change.

Reply:
Isn't it already breaking to add a new required parameter?

Reply:
> Isn't it already breaking to add a new required parameter?

Yes, but the fact that it didn't have this was a bug, so in that sense this breaking change is a "fixing change". It feels harder to argue that if we are completely changing the name.

That being said, if we are happy with breaking it, I'd rather break it that somehow try to preserve the buggy behavior as deprecated and create the solution side-by-side. It complicates the solution and just guarantees more future work in having to remove the old buggy code (again).

Reply:
I suppose I could preserve these two methods as deprecated but have the actual code use the new, correct logic. That way if there is an emitter that uses these methods (probably not, but I can't know for sure) then they'll continue to work.

Reply:
I preserved the old method and marked it deprecated, but none of the versioning cadl references the old name, so it only exists for emitters that might have been using it.

Reply:
I also had to do this for `getRenamedFromVersion` and create `getRenamedFromVersions`.